src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -62,9 +62,22 @@ public partial class MatchCollection : System.Collections.ICollection, System.Co
         void System.Collections.ICollection.CopyTo(System.Array array, int arrayIndex) { }
     }
     public delegate string MatchEvaluator(System.Text.RegularExpressions.Match match);
+    abstract public partial class RegexRunnerFactory
+    {
+        protected RegexRunnerFactory() { }
+        abstract protected internal RegexRunner CreateInstance();
+    }
     public partial class Regex
     {
         public static readonly System.TimeSpan InfiniteMatchTimeout;
+        protected internal string pattern;
```

Comment:
> I don't fully understand this reasoning - we are renaming the fields in the code from _pattern to pattern. This won't work in a backwards compatible manner, will it? If the fields are renamed, won't any existing assets need to be fixed?

These fields were renamed to prefix _ as part of open source code formatting, to make internal and private fields start with _. Since these are now public, and the field names are without _ in full desktop api set, these are going back to how they were before.

> Also, how come Caps and CapNames can be properties, but the rest of the values need to be fields?

Desktop has Hashtable for caps and capnames, the backing fields of these properties. And we dont want to take dependency on non generic collections, hence introducing new properties for these fields that takes a common IDictionary interface input, and gets cast to the desired types on desktop and netcore to hashtable and Dictionary respectively.